providers/class.two-factor-totp.php

```diff
@@ -377,10 +380,10 @@ public static function calc_totp( $key, $step_count = false, $digits = self::DEF
 	 */
 	public static function get_google_qr_code( $name, $key, $title = null ) {
 		// Encode to support spaces, question marks and other characters.
-		$name = rawurlencode( $name );
-		$google_url = urlencode( 'otpauth://totp/' . $name . '?secret=' . $key );
+		$name       = rawurlencode( $name );
+		$google_url = rawurlencode( 'otpauth://totp/' . $name . '?secret=' . $key );
 		if ( isset( $title ) ) {
-			$google_url .= urlencode( '&issuer=' . rawurlencode( $title ) );
+			$google_url .= rawurlencode( '&issuer=' . rawurlencode( $title ) );
```

Comment:
`$google_url` is going to be passed as a value to the `chl` query parameter so I think `urlencode()` might be needed here to ensure this doesn't break the current functionality per https://github.com/WordPress/WordPress-Coding-Standards/issues/1234#issuecomment-561435233

Reply:
From the following comment:
"At least for modern PHP, it seems that urldecode() can handle both, so there doesn't seem to be an obvious reason to use urlencode(), when rawlurlencode() follows an RFC (unless the generated URL is going to be sent to a system that has a different version of PHP, or has a method of decoding querystrings that doesn't support %20 for spaces)."

Reply:
> unless the generated URL is going to be sent to a system that has a different version of PHP, or has a method of decoding querystrings that doesn't support %20 for spaces.

We are sending this as a query string value to Google servers for generating the QR code and I remember there being issues exactly with how the query strings are parsed on their end.

In the worst case scenario the generated QR code will have an invalid authentication secret which will cause the generated codes to fail the authentication.

Could you maybe verify this locally just to confirm?